### PR TITLE
Adding terraform support and onbeforestart/onafterstart hooks

### PR DESCRIPTION
**Pull request by Julien4218**

### CODE REVIEW

File: src/deployment/aws/s3/validator.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            context
+          )
+          @services = services
+          @context = context
+        end
+
+        def execute(resources)
+          Common::Logger::LoggerFactory.get_logger().debug("Deployment/Aws/S3/Validator execute()")
+          validators = [
+            # No validators yet, add new rules here
```
Comment (gabeobrien) [inline src/deployment/aws/s3/validator.rb:19]: Are there any constraints on the bucket name we could validate?

Comment (gabeobrien) [reply src/deployment/aws/s3/validator.rb:19]: This is probably the wrong spot to really do this type of validation.  I missed it was part of 'deployment'.

Comment (Julien4218) [reply src/deployment/aws/s3/validator.rb:19]: I ll add a few rules following aws constraints.

Comment (Julien4218) [reply src/deployment/aws/s3/validator.rb]: @gabeobrien I've added a regex on the bucket name to validate the syntax (lower case, some numbers, with dot a dash allowed).

**Pull request merged**